R/step-group.R

```diff
@@ -1,17 +1,54 @@
-step_group <- function(parent, groups = parent$groups) {
+step_group <- function(parent, groups = parent$groups, keyed = parent$keyed) {
   new_step(
     parent,
     vars = parent$vars,
     groups = groups,
-    class = "dtplyr_step_group"
+    class = "dtplyr_step_group",
+    keyed = if (!is.null(keyed)) keyed else TRUE
   )
 }
 
+by_struct <- function(x) {
```

Comment:
I think this would be a bit simpler if you made it a function that modifies a call, something like this:

```R
add_by <- function(x, groups, keyed) {
  if (length(groups) == 0) {
    return(x)
  }
  
  using <- if (keyed) "keyby" else "by"
  call[[using]] <- call2(".", !!!syms(groups))
  
  call
}
```

The name isn't quite right, but I think that approach is a bit simpler.

Reply:
I agree! When I first started working on this, I thought the object under modification was a step, not the call itself, so I didn't want to introduce some weird side-effect. Since it's just a call, I removed the structure and replaced it with a function that's very similar to what you have written, which is in the latest push. I chose the name `add_grouping_parameter` for clarity.